Prox/Prox/Utilities/CategoriesUtil.swift

```diff
@@ -6,51 +6,81 @@ import Foundation
 
 struct CategoriesUtil {
 
-    static let HiddenRootCategories = Set(arrayLiteral: "auto", // 🚗
-                                                  "bicycles",
-                                                  "education",
-                                                  "financialservices",
-                                                  "health",
-                                                  "homeservices",
-                                                  "localservices",
-                                                  "professional",
-                                                  "realestate",
-                                                  "religiousorgs")
-
     private static let AllCategoriesPath = "Data.bundle/yelp_categories_v3"
     private static let AllCategoriesExt = "json"
 
     static func shouldShowPlace<S : Sequence>(byCategories categories: S) -> Bool where S.Iterator.Element == String {
-        // Hide if all match.
-        let rootCategories: Set<String>
-        do {
-            rootCategories = try getRootCategories(forCategories: categories)
-        } catch CategoryError.Unknown(let name) {
-            print("lol unknown category name, \(name) - filtering out by category")
-            return false
-        } catch { // I don't know why this is necessary - afaik, no other errors are thrown.
-            print("lol Unknown error occurred while filtering categories - filtering out")
-            return false
+        // TODO: UX still needs to tell us whether they want to hide a place if its categories contains
+        // a single category they said to hide or only if *all* its categories are in the categories
+        // they said to hide. For now, we hide if they all match.
+        let categorySet = Set(categories)
+        let allMatched = categorySet.subtracting(HiddenCategories).isEmpty
+        return !allMatched
+    }
+
+    // UX gives us a CSV list of categories to hide and we hide them *and their children*.
+    // This is an exhaustive collection of these categories and their children.
+    // Caveat: we don't expect UX to include root categories, which are special-cased: we can reduce
+    // yelp resource use by whitelisting the categories we ask for meaning the other root categories
+    // are hidden implicitly.
+    static let HiddenCategories: Set<String> = {
```

Comment:
nit: `hiddenCategories`. I know I've been bad at enforcing this but it's a new Swift 3.0 thing that constants are lowerCamelCase and it's taken me a while to get used to it.